File: app/src/main/java/studio/mandysa/music/service/model/AlbumModel.java
package studio.mandysa.music.service.model;

import java.util.ArrayList;

public interface AlbumModel {
    void add(MusicModel musicModel);

    void remove(MusicModel musicModel);

    int size();

    MusicModel get(int index);

    ArrayList<ArtistModel> getArtist();
}

File: media/src/main/java/mandysax/media/model/AlbumModel.kt
package mandysax.media.model

import java.util.*

/**
 * @author liuxiaoliu66
 */
interface AlbumModel<T : MusicModel<out ArtistModel>> {
    fun add(vararg musicModels: DefaultMusic<DefaultArtist>)
    fun remove(vararg musicModels: T)
    fun size(): Int
    operator fun get(index: Int): T
    val artist: ArrayList<*>?
}